packages/channel-client/tests/fake-channel-client.test.ts

```diff
@@ -0,0 +1,184 @@
+import {ChannelResult, Message} from '../src';
+import {EventsWithArgs} from '../src/types';
+import {calculateChannelId} from '../src/utils';
+
+log.setDefaultLevel(log.levels.SILENT);
+
+interface StateMap {
+  [channelStatus: string]: ChannelResult;
+}
+
+describe('FakeChannelClient', () => {
+  const participantA = buildParticipant(PARTICIPANT_A);
+  const participantB = buildParticipant(PARTICIPANT_B);
+
+  const states: StateMap = {};
+  const participants = [participantA, participantB];
+  const allocations = [buildAllocation(PARTICIPANT_A, '5'), buildAllocation(PARTICIPANT_B, '5')];
+  const channelId = calculateChannelId(participants, allocations, APP_DEFINITION, APP_DATA);
+
+  // This event emitter only enable easier assertions of expected
+  // events the client is listening on
+  const clientBEventEmitter = new EventEmitter<EventsWithArgs>();
+
+  let clientA: FakeChannelClient, clientB: FakeChannelClient;
+
+  beforeAll(() => {
+    states['proposed'] = new ChannelResultBuilder(
+      participants,
+      allocations,
+      APP_DEFINITION,
+      APP_DATA,
+      channelId,
+      '0',
+      'proposed'
+    ).build();
+
+    states['running'] = ChannelResultBuilder.from(states['proposed'])
+      .setStatus('running')
+      .setTurnNum('3')
+      .build();
+
+    states['updated_app_data'] = ChannelResultBuilder.from(states['running'])
+      .setAppData(UPDATED_APP_DATA)
+      .setTurnNum('4')
+      .build();
+
+    states['closed'] = ChannelResultBuilder.from(states['running'])
+      .setStatus('closed')
+      .setTurnNum('5')
+      .build();
+  });
+
+  beforeEach(() => {
+    clientA = new FakeChannelClient(participantA.participantId);
+    clientB = new FakeChannelClient(participantB.participantId);
+
+    clientA.updatePlayerIndex(0);
+    clientA.opponentAddress = participantB.participantId;
+
+    clientB.updatePlayerIndex(1);
+    clientB.opponentAddress = participantA.participantId;
+
+    // This setup simulates the message being received from A's wallet
+    // and "queued" by A's app to be sent to the opponent (handled by
+    // A's channel client, which then is "dequeued" on A's channel client
+    // and sent to B's app (handled by B's channel client here) and
+    // pushed from B's app to B's wallet.
+    // The de/queuing described above is effectively faked by explicitly passing
+    // the messages between the clients.
+    clientA.onMessageQueued(async (message: Message<ChannelResult>) => {
+      await clientB.pushMessage(message);
+    });
+
+    clientB.onMessageQueued(async (message: Message<ChannelResult>) => {
+      await clientA.pushMessage(message);
+    });
+
+    clientB.onChannelProposed((result: ChannelResult) => {
+      clientBEventEmitter.emit('ChannelProposed', result);
+    });
+  });
+
+  it('instantiates', () => {
+    expect(clientA).toBeDefined();
+    expect(clientB).toBeDefined();
+  });
```

Comment:
I think this test is redundant because `beforeEach` would explode if neither were defined.